cmd/samples/cron/main.go

```diff
@@ -1,66 +1,95 @@
 package main
 
 import (
+	"context"
 	"flag"
 	"time"
 
 	"github.com/pborman/uuid"
 	"go.temporal.io/temporal/client"
 	"go.temporal.io/temporal/worker"
-
-	"github.com/temporalio/temporal-go-samples/cmd/samples/common"
+	"go.uber.org/zap"
 )
 
 const (
-	// ApplicationName is the task list for this sample
-	ApplicationName = "cronGroup"
+	// DomainName is the name of domain where workflow will be created.
+	DomainName = "samples-domain"
+
+	// TaskListName is the task list for this sample.
+	TaskListName = "cron-task-list"
+)
+
+var (
+	logger *zap.Logger
 )
 
+func main() {
+	var err error
+	logger, err = zap.NewDevelopment()
+	if err != nil {
+		panic(err)
+	}
+
+	var mode string
+	var cron string
+	flag.StringVar(&mode, "m", "trigger", "Mode is worker or trigger.")
+	flag.StringVar(&cron, "cron", "* * * * *", "Crontab schedule. Default \"* * * * *\"")
+	flag.Parse()
+
+	switch mode {
+	case "worker":
+		startWorker()
+		// The workers are supposed to be long running process that should not exit.
+		// Use select{} to block indefinitely for samples, you can quit by Ctrl+C.
+		select {}
+	case "trigger":
+		startWorkflow(cron)
+	}
+}
+
 // This needs to be done as part of a bootstrap step when the process starts.
 // The workers are supposed to be long running.
-func startWorkers(h *common.SampleHelper) {
-	// Configure worker options.
+func startWorker() {
 	workerOptions := worker.Options{
-		MetricsScope: h.Scope,
-		Logger:       h.Logger,
+		Logger: logger,
```

Comment:
How do we configure hostport?  If someone wants to run samples against a real deployment of Temporal.

Reply:
I don't want to use config because it adds complexity to the code and restriction to how we run samples. I can pass default value `localhost:7233` using options and it will allow users to change it easily.

Reply:
I've just created `client.DefaultHostPort` const in SDK. I will pass it explicitly here.